### ISSUE TITLE
Create the description of the planned software architecture

### ISSUE THREAD

**Issue by Nahrain1** [amosproj/amos2024ss07-updating-flash-boot-loader#14]
## User story
1. As a Software Developer
2. I want to prepare a detailed description of the planned software architecture
3. So that we have an overview of the system's design, including runtime and code components, technology stack, and the rationale behind architectural choices.

## Acceptance criteria
* An overview diagram of runtime components (one page) is created
* An overview diagram of code components (one page) is created
* A summary of the underlying technology stack (one page) is created
* A textual explanation of the diagrams and choices (one page) is created
* All documents are merged to one PDF file
* PDF is submitted to POs on time (on slack for example) so that they can upload the document before the deadline

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
